modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/impl/PageMemoryNoStoreImpl.java

```diff
@@ -167,23 +169,26 @@ public class PageMemoryNoStoreImpl implements PageMemory {
     /**
      * Constructor.
      *
-     * @param directMemoryProvider Memory allocator to use.
      * @param dataRegionConfig Data region configuration.
      * @param ioRegistry IO registry.
      * @param pageSize Page size in bytes.
      */
     public PageMemoryNoStoreImpl(
-            DirectMemoryProvider directMemoryProvider,
             PageMemoryDataRegionConfiguration dataRegionConfig,
             PageIoRegistry ioRegistry,
             // TODO: IGNITE-17017 Move to common config
             int pageSize
     ) {
-        this.directMemoryProvider = directMemoryProvider;
         this.ioRegistry = ioRegistry;
         this.trackAcquiredPages = false;
         this.dataRegionConfigView = dataRegionConfig.value();
 
+        if (dataRegionConfigView.memoryAllocator() instanceof UnsafeMemoryAllocator) {
+            throw new IgniteInternalException("Unexpected memory allocator: " + dataRegionConfigView.memoryAllocator());
```

Comment:
Do we actually have "toString()" implementation of configuration nodes? I guess we should render them in HOCON, for example, what do you think?

Reply:
I think at the beginning it is necessary to define the general format for errors related to the configuration, and then we will redo it.

Reply:
Ok, I guess we will literally go through every throw in our codebase, is that correct?

Reply:
The whole configuration in a cocoon? one line? with separators for human reading? How to highlight that this particular configuration among all the roots is erroneous? what should the user do?